giving-example/src/main/java/com/adyen/giving/api/CheckoutResource.java

```diff
@@ -133,6 +149,16 @@ public ResponseEntity<PaymentResponse> payments(@RequestHeader String host, @Req
 
         log.info("REST request to make Adyen payment {}", paymentRequest);
         var response = paymentsApi.payments(paymentRequest);
+
+        var session = request.getSession();
+        session.setAttribute(paymentOriginalPspReference, response.getPspReference());
```

Comment:
I wouldn't add anything in Session if token is unavailable: add both values or nothing is better IMO